File: backened/utils/promptBuilder.js
export function buildPrompt({
  text,
  numMcqs = 10,
  difficulty = "medium",
  chapterTitles = [],
  summaryType = "normal",
  language = "en",
  generateNotes = true,
  generateMcqs = true,
}) {
  const noteInstruction = (() => {
    switch (summaryType) {
      case "quick":
        return "Notes MUST consist of exactly 5 concise, high-impact, exam-focused bullet points (using • or - for each point). Keep it very brief.";
      case "deep":
        return "Notes MUST be highly detailed, explaining key concepts thoroughly, defining terminology, and including illustrative examples.";
      default:
        return "Notes MUST be well-structured, clear, concise, and exam-focused study guides.";
    }
  })();

  return `
You are an expert educational examiner and academic writer.
Your task is to analyze the provided study material and generate
high-quality exam preparation content.

PARAMETERS:
- MCQ count: ${numMcqs}
- Difficulty: ${difficulty} (easy=recall, medium=understanding, hard=application/analysis)
- Summary type: ${summaryType}
- Chapters: ${chapterTitles.join(", ") || "Full document"}
- Output language: ${language}
- Generate notes: ${generateNotes}
- Generate MCQs: ${generateMcqs}

STRICT RULES:
- MCQ wrong options must be plausible — not obviously wrong
- Questions must test understanding, not just memorization
- Each question must have exactly one unambiguous correct answer
- Notes format rule: ${noteInstruction}
- Do NOT wrap the JSON response in markdown code blocks (do not use \`\`\`json ... \`\`\` formatting). Respond ONLY with the raw, valid JSON object.
- Within the JSON string fields, you may use standard punctuation and formatting (such as bullet points, hyphens, or newlines) to format the notes nicely.

REQUIRED JSON FORMAT (respond with this exact structure):
{
  "mcqs": [
    {
      "question": "...",
      "options": ["A. ...", "B. ...", "C. ...", "D. ..."],
      "answer": "A",
      "explanation": "Why this answer is correct and others are wrong",
      "topic": "Chapter or topic this question covers"
    }
  ],
  "notes": [
    {
      "heading": "Topic Title",
      "content": "Detailed exam-focused explanation"
    }
  ],
  "detectedLanguage": "en",
  "difficulty": "${difficulty}"
}

STUDY MATERIAL:
${text}
`.trim();
}

File: ai_learning_app/lib/services/prompt_builder.dart
class PromptBuilder {
  static String build({
    required String text,
    required int numMcqs,
    required String difficulty,
    required bool generateNotes,
    required bool generateMcqs,
    String summaryType = 'normal',
  }) {
    final noteInstruction = switch (summaryType) {
      'quick' => 'Create 5 concise, exam-focused bullet sections.',
      'deep' => 'Create detailed study notes with key terms and short examples.',
      _ => 'Create concise, well-structured, exam-focused study notes.',
    };

    return '''You are an expert educational writer. Analyze only the study material below.

OUTPUT REQUEST:
- Generate notes: $generateNotes
- Generate MCQs: $generateMcqs
${generateNotes ? '- Notes: $noteInstruction' : ''}
${generateMcqs ? '- Generate exactly $numMcqs $difficulty-difficulty MCQs. Each question must have four plausible options and one unambiguous answer.' : ''}

Return valid JSON only. Do not include markdown or commentary.
Use empty arrays for content that was not requested.

JSON FORMAT:
{
  "mcqs": [
    {
      "question": "...",
      "options": ["A. ...", "B. ...", "C. ...", "D. ..."],
      "answer": "A",
      "explanation": "...",
      "topic": "..."
    }
  ],
  "notes": [
    {
      "heading": "...",
      "content": "..."
    }
  ]
}

STUDY MATERIAL:
$text''';
  }
}
